### ISSUE TITLE
The example query-builder.js not completing

### ISSUE THREAD

**Issue by wooldridge** [marklogic/node-client-api#362]
When running all the examples at once, as described in the README, or when running query-builder.js alone, execution dies near the end of query-builder.js:
```
mwooldri$ node examples/query-builder.js 
Query documents with the Query Builder
Mayotte at /countries/mf.json
Seychelles at /countries/se.json
Morocco at /countries/mo.json
Burkina Faso at /countries/uv.json
Algeria at /countries/ag.json
Djibouti at /countries/dj.json
Cote d'Ivoire at /countries/iv.json
Mauritania at /countries/mr.json
Guinea at /countries/gv.json
Mali at /countries/ml.json
mwooldri$
```
It should end like this:
```
...
Mali at /countries/ml.json
Niger at /countries/ng.json
done
mwooldri$ 
```
All other examples complete successfully when run by themselves.

When query-builder.js is edited to use a promise instead of a stream (similar to how query-by-example.js uses a promise), the script completes successfully. So it appears to be an issue with streams in query-builder.js. 

I'm running Node.js v6.5.0 and npm v3.10.3.

**Comment by wooldridge** [comment]
@ayuwono, with this fix we've refactored how we check if a multipart object stream is finished. Can you add the following test to check that this works with all cases?

Get results that have two documents with *metadata only* (no content) as an *object stream*. (This will test the situation when metadata is the last part in the multipart queue.)

Thanks to @ehennum for his help on this!

**Comment by ayuwono** [comment]
Hi @wooldridge,

Is this fixed? I tried to run the example query-builder.js and it's still not completing

```
[ayuwono@ayuwono-z620 node-client-api]$ node examples/query-builder.js
Query documents with the Query Builder
Mayotte at /countries/mf.json
Seychelles at /countries/se.json
Morocco at /countries/mo.json
Burkina Faso at /countries/uv.json
Algeria at /countries/ag.json
Djibouti at /countries/dj.json
Cote d'Ivoire at /countries/iv.json
Mauritania at /countries/mr.json
Guinea at /countries/gv.json
Mali at /countries/ml.json
[ayuwono@ayuwono-z620 node-client-api]$
```

**Comment by wooldridge** [comment]
Update, the query-builder.js example should run like this, returning the first 10 results, so ending at "Mali at /countries/ml.json" followed by "done":
```
MacPro-3170:node-client-api-examples mwooldri$ node examples/query-builder.js 
Query documents with the Query Builder
Mayotte at /countries/mf.json
Seychelles at /countries/se.json
Morocco at /countries/mo.json
Burkina Faso at /countries/uv.json
Algeria at /countries/ag.json
Djibouti at /countries/dj.json
Cote d'Ivoire at /countries/iv.json
Mauritania at /countries/mr.json
Guinea at /countries/gv.json
Mali at /countries/ml.json
done
MacPro-3170:node-client-api-examples mwooldri$ 
```

**Comment by gvaidees** [comment]
@wooldridge This is targeted for 9.0-2. Would you be able to take care of this?

**Comment by wooldridge** [comment]
@gvaidees Thanks for the reminder. This was targeted for 9.0-2 but @ayuwono and I were having trouble seeing the fix resolve on Linux. So I am going to push it out to 9.0-3.

**Comment by wooldridge** [comment]
Should be fixed by #422. @ayuwono, needs QA.
